packages/backend/src/nest/storage/storage.service.ts

```diff
@@ -419,6 +423,25 @@ export class StorageService extends EventEmitter {
     this.logger('STORAGE: Finished creating certificatesRequests db')
   }
 
+  public async filterCsrs(allCsrs: string[]): Promise<string[]> {
+    const filteredCsrsMap: Map<string, string> = new Map()
+
+    await Promise.all(
+      allCsrs.map(async csr => {
+        const parsedCsr = await loadCSR(csr)
+        const pubKey = keyFromCertificate(parsedCsr)
+
+        if (filteredCsrsMap.has(pubKey)) {
+          filteredCsrsMap.delete(pubKey)
+        }
+
+        filteredCsrsMap.set(pubKey, csr)
+      })
+    )
+
+    return [...filteredCsrsMap.values()]
```

Comment:
I think we can just do ```...filteredCsrsMap.values()```

Reply:
Don't you mean `filteredCsrsMap.values()`?

Reply:
I want to return csrs  in that format `[val1, val2, val3...]`

Reply:
Yes, sorry, I meant `filteredCsrsMap.values()` and sorry again, I was looking at Object.values (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/values) which returns an array. Map.values returns an iterator (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map/values). So the initial line looks correct.